search-service/src/main/kotlin/com/egm/stellio/search/listener/IAMListener.kt

```diff
@@ -99,9 +105,19 @@ class IAMListener(
         tenantName: String,
         entityDeleteEvent: EntityDeleteEvent
     ): Either<APIException, Unit> = either {
+        val subjectType = SubjectType.valueOf(entityDeleteEvent.entityTypes.first().uppercase())
         mono {
             subjectReferentialService.delete(entityDeleteEvent.entityId.extractSub())
         }.writeContextAndSubscribe(tenantName, entityDeleteEvent)
+        if (searchProperties.deleteEntityByCreator && subjectType == SubjectType.USER) {
+            val entitiesIds = entityAccessRightsService.getEntitiesIdsOwnedBySubject(
+                entityDeleteEvent.entityId.extractSub()
```

Comment:
I had difficutlies understanding this part maybe put```entityDeleteEvent.entityId.extractSub()``` in a deletedUserId variable, to clarify what it is.